src/Data/Rational/Unnormalised.agda

```diff
@@ -83,9 +86,58 @@ x ≮ y = ¬ (x < y)
 _≯_ : Rel ℚᵘ 0ℓ
 x ≯ y = ¬ (x > y)
 
+------------------------------------------------------------------------
+-- Non-negative and negative rationals
+
+record non-neg (p : ℚᵘ) : Set where
```

Comment:
Hmm so I've already had a go at formalizing such predicates in #850 in a compositional manner for all the numeric types. I had some concerns over instance arguments and I'll try remember what they are and update and merge in the PR later this week.